### PR TITLE
Update styling mobile image-only logo #1466

### PR DESCRIPTION
**Pull request by akegan**
## Issue #1466 

Addresses # (issue)

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR helps us get the logo looking better for DAHLIA.

It:
1. Uses our SVG in the story rather than the listing image. This helped a lot w/ being able to visualize the implementation, and is more of a real-world example, but I can switch it back to the listing image if you'd like
2. Updates the default mobile height for the image-only component so that the logo/menu bar is 75px tall. The issue originally wanted to make this configurable, but that would get complicated w/ setting the medium-up version at a height of 120px. It would have required creating new tailwind utility classes, which didn't seem worthwhile given that we are the only current use-case and it seems like we just didn't start with a reasonable default.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Look at `/story/headers-site-header--image-only-medium` component at different screen sizes to see if the new height is a reasonable default. 

- [ ] Desktop View
- [ ] Mobile View

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: ui-components/public/images/DAHLIA-logo.svg [hunk 1]
```diff
@@ -0,0 +1,155 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (emilyjablonski) [inline ui-components/public/images/DAHLIA-logo.svg:1]: It seems odd to me to add a DAHLIA-specific logo to ui-components, was this intentional? Just for testing purposes?

Comment (emilyjablonski) [reply ui-components/public/images/DAHLIA-logo.svg:1]: Oop seeing your note in the PR description now - I would personally prefer to keep the logo as it was or to add a generic one (but that seems like more effort than it is worth)

Comment (akegan) [reply ui-components/public/images/DAHLIA-logo.svg:1]: yeah it just seemed like a more realistic example than the listing image 😕 . I can update to put it back to the listing image now that I've tested it out

File: ui-components/src/global/navbar.scss [hunk 2]
```diff
@@ -61,7 +61,8 @@ nav.navbar {
 }
 
 .navbar-image-only-container {
-  height: 50px;
+  // 75px - .5rem of padding from the container around it.
+  height: 67px;
```
Comment (akegan) [inline ui-components/src/global/navbar.scss:65]: I realize this seems like a magic odd number. I'm up for rounding up to 4.25rem or 75px here if you prefer. 75px for the total height just looked nice, but we can be a bit flexible

File: ui-components/public/images/DAHLIA-logo.svg (hunk 1 above)
Comment (emilyjablonski) [reply ui-components/public/images/DAHLIA-logo.svg:1]: I'll make a ticket on our end to get some better generic testing SVGs in there :)

Comment (akegan) [reply ui-components/public/images/DAHLIA-logo.svg]: pushed!

**Review by emilyjablonski — APPROVED** [review]

**Pull request merged**